File: ruby/grade-school/grade_school.rb
class School
  attr_reader :grades

  def initialize
    @grades = {}
  end

  def add(name, grade)
    grades[grade] ? grades[grade] << name : grades[grade] = [name]
  end

  def grade(num)
    grades[num] ? grades[num].sort : []
  end

  def to_hash
    grades.sort.each_with_object({}) do |(grade, names), result|
      result[grade] = names.sort
    end
  end
end

File: lua/grade-school/grade-school.lua
local School = {}

function School:new()
  self.students = {}
  return self
end

function School:roster()
  table.sort(self.students)
  for grade, _ in pairs(self.students) do table.sort(self.students[grade]) end
  return self.students
end

function School:add(name, grade)
  if self.students[grade] then
    table.insert(self.students[grade], name)
  else
    self.students[grade] = { name }
  end
end

function School:grade(grade)
  if self.students[grade] then
    table.sort(self.students[grade])
  else
    self.students[grade] = {}
  end
  return self.students[grade]
end

return School

